### PR TITLE
Preserve zero decimal (backport to 2.8.x)

### PR DESCRIPTION
**Pull request by trbogart**
# Pull Request Checklist

* [X] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [X] Have you [squashed your commits](https://www.playframework.com/documentation/latest/WorkingWithGit#Squashing-commits)?
* [X] Have you added copyright headers to new files?
* [x] Have you updated the documentation?
 - Updated documentation in code
* [X] Have you added tests for any changed functionality?

## Fixes

See https://github.com/playframework/play-json/pull/831 for more information. This backports that to 2.8.x.

### CODE REVIEW

**Comment by mkurz** [comment]
Not sure why GHA is not running, we do have a workflow for the 2.9.x branch: https://github.com/playframework/play-json/blob/2.9.x/.github/workflows/build-test.yml

**Comment by mkurz** [comment]
Ups, sorry this is the 2.8.x PR, my bad, we don't have workflow here. We can test locally :wink:

**Comment by mkurz** [comment]
I ran the tests locally like we did with Travis back then:
https://github.com/playframework/play-json/blob/b82966d7b523410f55e52b62811609765ceccb25/.travis.yml#L17-L19

All tests pass, I only fixed a header which I pushed, and we get the same mima errors like in https://github.com/playframework/play-json/blob/b82966d7b523410f55e52b62811609765ceccb25/.travis.yml#L17-L19 so actually we should be ok to merge this.

@trbogart Can you please add the same mima filters like in the other pull request? I will then run the tests locally again and see if all passes. Thanks!

**Comment by trbogart** [comment]
> @trbogart Can you please add the same mima filters like in the other pull request? I will then run the tests locally again and see if all passes. Thanks!

Done. 

BTW, I can't test this or do anything with this branch locally because of the following error:
```
[error] sbt.AutoPluginException: Error determining plugins for project 'play-jsonJVM' in /Users/richard.bogart/code/playframework/play-json/play-json/jvm:
[error] Atom(Playdoc) was not found in atom map.
```

File: play-json/jvm/src/main/scala/play/api/libs/json/JsonConfig.scala [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      case NonFatal(_) => default
+    }
+}
+
+private final case class JsonConfigImpl(
+    bigDecimalParseConfig: BigDecimalParseConfig,
+    bigDecimalSerializerConfig: BigDecimalSerializerConfig
+) extends JsonConfig
+
+@deprecated("Use BigDecimalParseConfig instead", "2.8.3")
```
Comment (trbogart) [inline play-json/jvm/src/main/scala/play/api/libs/json/JsonConfig.scala:226]: Update deprecated version tags

Comment (mkurz) [reply play-json/jvm/src/main/scala/play/api/libs/json/JsonConfig.scala:226]: Makes sense.

**Comment by mkurz** [comment]
For testing, I setup a local bash script, which just resembles what's in the `.travis.yml` config in the 2.8.x branch: 
https://github.com/playframework/play-json/blob/b82966d7b523410f55e52b62811609765ceccb25/.travis.yml#L18-L19

So that's what I do:

```
$ java -version
openjdk version "1.8.0_352"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_352-b08)
OpenJDK 64-Bit Server VM (Temurin)(build 25.352-b08, mixed mode)

$ cat ./test.sh 
#!/usr/bin/env bash

run_tests () {
  sbt -DscalaJSStage=full ++$TRAVIS_SCALA_VERSION validateCode test mimaReportBinaryIssues publishLocal
  sbt ++$TRAVIS_SCALA_VERSION docs/test docs/validateDocs
}

TRAVIS_SCALA_VERSION=2.12.10
run_tests
TRAVIS_SCALA_VERSION=2.13.1
run_tests

$ ./test.sh
# ...runs for ~ 3 minutes...
# All tests pass :)
```

Just did this with your latest commits you pushed to this PR, looks good :+1: 

@gmethvin Please merge if you think it's good to go, specially the MiMa filters, but as you said they should just filter private things. Like said, all tests and checks pass locally for me.

**Comment by trbogart** [comment]
We ended up upgrading to 2.9.x, so we don't actually need this.

**Comment by mkurz** [comment]
Yeah I think it's also safter to not merge this into the 2.8.x branch.

**Pull request closed without merging**